compositor_pipeline/src/pipeline/output/rtp/payloader.rs

```diff
@@ -0,0 +1,228 @@
+    pub fn new(
+        video: Option<(VideoCodec, PayloadType)>,
+        audio: Option<(AudioCodec, PayloadType)>,
+    ) -> Self {
+        Self {
+            video: video.map(|(codec, payload_type)| VideoPayloader::new(codec, payload_type)),
+            audio: audio.map(|(codec, payload_type)| AudioPayloader::new(codec, payload_type)),
+        }
+    }
+
+    pub fn payload(
+        &mut self,
+        mtu: usize,
+        data: EncodedChunk,
+    ) -> Result<Vec<rtp::packet::Packet>, PayloadingError> {
+        match data.kind {
+            EncodedChunkKind::Video(chunk_codec) => {
+                let Some(ref mut video_payloader) = self.video else {
+                    return Err(PayloadingError::NoVideoPayloader);
+                };
+
+                if video_payloader.codec() != chunk_codec {
+                    return Err(PayloadingError::NonMatchingVideoCodecs {
+                        chunk_codec,
+                        payloader_codec: video_payloader.codec(),
+                    });
+                }
+
+                video_payloader.payload(mtu, data)
+            }
+            EncodedChunkKind::Audio(chunk_codec) => {
+                let Some(ref mut audio_payloader) = self.audio else {
+                    return Err(PayloadingError::NoAudioPayloader);
+                };
+
+                if audio_payloader.codec() != chunk_codec {
+                    return Err(PayloadingError::NonMatchingAudioCodecs {
+                        chunk_codec,
+                        payloader_codec: audio_payloader.codec(),
+                    });
+                }
+
+                audio_payloader.payload(mtu, data)
+            }
+        }
+    }
+}
+
+impl VideoPayloader {
+    pub fn new(codec: VideoCodec, payload_type: PayloadType) -> Self {
+        match codec {
+            VideoCodec::H264 => Self::H264 {
+                payloader: H264Payloader::default(),
+                payload_type,
+                context: RtpStreamContext::new(),
+            },
+        }
+    }
+
+    pub fn codec(&self) -> VideoCodec {
+        match self {
+            VideoPayloader::H264 { .. } => VideoCodec::H264,
+        }
+    }
+
+    pub fn payload(
+        &mut self,
+        mtu: usize,
+        chunk: EncodedChunk,
+    ) -> Result<Vec<rtp::packet::Packet>, PayloadingError> {
+        match self {
+            VideoPayloader::H264 {
+                ref mut payloader,
+                ref payload_type,
+                ref mut context,
+            } => payload(payloader, context, chunk, mtu, payload_type),
+        }
+    }
+}
+
+impl AudioPayloader {
+    pub fn new(codec: AudioCodec, payload_type: PayloadType) -> Self {
+        match codec {
+            AudioCodec::Opus => Self::Opus {
+                payloader: OpusPayloader,
+                payload_type,
+                context: RtpStreamContext::new(),
+            },
+            AudioCodec::Aac => panic!("Aac audio output is not supported yet"),
```

Comment:
This should never happen. Guaranteeing it on the type system would require creating another `VideoCodec` type, so I decided to leave it this way